scripts/calculateRevenue/protocols/aave/index.ts

```diff
@@ -41,8 +66,202 @@ export async function revenueInNetwork(
     endTimestamp,
   )
 
-  // TODO: Implement revenue calculation logic
-  console.log('chainData', chainData)
+  const context: RevenueCalculationContext = {
+    ...chainData,
+    startTimestamp: Math.floor(startTimestamp.getTime() / 1000),
+    endTimestamp: Math.floor(endTimestamp.getTime() / 1000),
+  }
+
+  const protocolRevenueByReserve = revenueByReserve(context)
+
+  const revenue = totalRevenueInUSD(protocolRevenueByReserve, context)
+
+  return revenue
+}
+
+function revenueByReserve(context: RevenueCalculationContext): Revenue[] {
+  return [...context.endReserveData.values()].map(
+    ({ reserveTokenAddress, reserveTokenDecimals, aTokenAddress }) =>
+      revenueInReserve(
+        reserveTokenAddress,
+        reserveTokenDecimals,
+        aTokenAddress,
+        context,
+      ),
+  )
+}
+
+export function revenueInReserve(
```

Comment:
nit: are we only exporting this because of unit tests? I understand it's convenient, but it's usually preferable to test only the public interface of a module.